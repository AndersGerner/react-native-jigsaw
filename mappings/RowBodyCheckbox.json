[{"name":"Row Single Line Body Checkbox","tag":"RowBodyCheckbox","description":"A row with left aligned body text and a right aligned checkbox","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_SingleLine_BodyCheckbox.png","category":"deprecated","supports_list_render":true,"props":{"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"color":{"label":"Checkbox color","description":"Custom color for Checkbox","editable":true,"required":false,"type":"color"},"fieldName":{"label":"Field name","description":"The name of the field within the screen that will store this component's value","type":"fieldName","value":"checkboxValue","valuePropName":"status","handlerPropName":"onPress","editable":true,"required":false}},"layout":{}},{"name":"Row Double Line Body Checkbox","tag":"RowBodyCheckbox","description":"A row with left aligned body text and subtitle text and a right aligned checkbox","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_DoubleLine_BodyCheckbox.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"subtitle":{"label":"Subtitle","description":"Subtitle text to display","type":"string","value":"San Diego","editable":true},"color":{"label":"Color","description":"Custom color for Checkbox","editable":true,"required":false,"type":"color"},"fieldName":{"label":"Field name","description":"The name of the field within the screen that will store this component's value","type":"fieldName","value":"checkboxValue","valuePropName":"status","handlerPropName":"onPress","editable":true,"required":false}},"layout":{}}]