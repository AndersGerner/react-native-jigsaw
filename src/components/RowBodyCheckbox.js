/* @flow */
import * as React from "react";
import { withTheme } from "../core/theming";
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types";
import Row from "./Row";
import Checkbox from "./Checkbox";
import Config from "./Config";
import type { Theme } from "../types";

type Props = {
  title: string,
  subtitle?: string,
  style?: any,
  theme: Theme,
  status: "checked" | "unchecked" | "indeterminate",
  color?: string,
  onPress?: () => mixed
};

class RowBodyCheckbox extends React.Component<Props> {
  render() {
    const {
      title,
      subtitle,
      style,
      status,
      onPress,
      color,
      theme: { colors, typography }
    } = this.props;

    return (
      <Row
        titleTypeStyle={typography.body1}
        titleColor={colors.medium}
        subtitleTypeStyle={typography.subtitle2}
        subtitleColor={colors.light}
        title={title}
        subtitle={subtitle}
        right={() => (
          <Checkbox
            status={status}
            size={Config.rowSingleLineIconSize}
            color={color}
            onPress={onPress}
          />
        )}
        style={style}
      />
    );
  }
}

export default withTheme(RowBodyCheckbox);

export const SEED_DATA = [
  {
    name: "Row Single Line Body Checkbox",
    tag: "RowBodyCheckbox",
    description:
      "A row with left aligned body text and a right aligned checkbox",
    preview_image_url:
      "{CLOUDINARY_URL}/Row_SingleLine_BodyCheckbox.png",
    category: COMPONENT_TYPES.row,
    supports_list_render: true,
    props: {
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      status: {
        label: "Check status",
        description: "Status of checkbox",
        required: true,
        editable: true,
        value: "checked",
        type: FORM_TYPES.flatArray,
        options: ["checked", "unchecked", "indeterminate"]
      },
      color: {
        label: "Checkbox color",
        description: "Custom color for Checkbox",
        editable: true,
        required: false,
        type: FORM_TYPES.color
      },
      onPress: {
        label: "OnPress handler",
        description: "function that handles checking or unchecking",
        editable: true,
        required: false,
        value: "{this.onPress}",
        type: FORM_TYPES.function
      }
    },
    layout: {
      width: 375,
      height: 59
    }
  },
  {
    name: "Row Double Line Body Checkbox",
    tag: "RowBodyCheckbox",
    description:
      "A row with left aligned body text and subtitle text and a right aligned checkbox",
    category: COMPONENT_TYPES.row,
    preview_image_url:
      "{CLOUDINARY_URL}/Row_DoubleLine_BodyCheckbox.png",
    supports_list_render: true,
    props: {
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      subtitle: {
        label: "Subtitle",
        description: "Subtitle text to display",
        type: FORM_TYPES.string,
        value: "San Diego",
        editable: true
      },
      status: {
        label: "Status",
        description: "Status of checkbox",
        required: true,
        editable: true,
        value: "checked",
        type: FORM_TYPES.flatArray,
        options: ["checked", "unchecked", "indeterminate"]
      },
      color: {
        label: "Color",
        description: "Custom color for Checkbox",
        editable: true,
        required: false,
        type: FORM_TYPES.color
      },
      onPress: {
        label: "onPress handler",
        description: "function that handles checking or unchecking",
        editable: true,
        required: false,
        value: "{this.onPress}",
        type: FORM_TYPES.function
      }
    },
    layout: {
      width: 375,
      height: 81
    }
  }
];
