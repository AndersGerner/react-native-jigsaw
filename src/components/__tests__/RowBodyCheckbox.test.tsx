import * as React from "react";
import renderer from "react-test-renderer";
import RowBodyCheckbox from "../RowBodyCheckbox";

it("renders row body checkbox", () => {
  const tree = renderer.create(<RowBodyCheckbox />).toJSON();

  expect(tree).toMatchSnapshot();
});
