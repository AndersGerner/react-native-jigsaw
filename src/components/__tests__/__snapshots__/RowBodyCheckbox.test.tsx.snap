// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders row body checkbox 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 26,
            },
            Object {
              "color": "rgba(70, 78, 88, 1)",
            },
          ]
        }
      />
    </View>
  </View>
  <View
    accessibilityLiveRegion="polite"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "borderColor": "rgba(165, 173, 183, 1)",
        "borderRadius": 2,
        "borderWidth": 2,
        "height": 25,
        "opacity": 1,
        "width": 25,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
